rfcs/001 - EC2 schema.md

```diff
@@ -0,0 +1,110 @@
+
+We can divide the feature list as follow:
+
+The direct implication in the `instance` table:
+
+- Launch templates
+- Spot instances
+- Dedicated hosts
+- Capacity reservations
+- Auto Scaling Groups
+- Elastic IP integration
+- Placement groups
+- Network interfaces
+
+No implicaiton in the `instance` table:
+
+- Instance type - Availability zone relationship
+- AMIs management
+- EBS volume snapshots and life cycle
+
+The `Saving plans`, `Reserved instances` or `Scheduled instances` are not included in the list above since they seem to be budgeting solutions that need interaction with the user rather than infrastructure-related modules' features that could have any implication in the model.
+
+Let's start with the features without impact on the `instance` table. These features will be part of the `aws_ec2` module with their mappers. The Instance type table by availability zones will be a table with a composite primary key. It technically has a minimum impact in the `instance` table that will be a special FK to the composite primary key, but no UX model impact.
+
+The features with direct implications in the `instance` table can be divided into two groups. The first group will not only affect the `aws_ec2` module, and the second group will.
+
+Let's start with the first group, Placement groups and Network interfaces. These features will be part of their respective modules (probably `aws_vpc`?). Then, they will need to be integrated with the `instance` table in the `aws_ec2` module with an FK nullable column.
+
+The rest of the features will need to be implemented in the `aws_ec2` module and integrated with the `instance` table in the `aws_ec2` module with an FK nullable column.
+
+Spot instances and Auto Scaling Group can be updated at any point in time by the cloud. These will imply the development of the "safer sync faster apply" feature as a pre-requisite for the two.
+
+Launch templates, auto-scaling group requests and spot instances requests can be linked to an instance using the tags AWS assign with a specific pattern. Otherwise will be hard to link all the relationships due to the async nature of ASG and spot instances.
+
+### Alternatives Considered
+
+Create different modules for the services that can manage AWS instances.
+Create different tables within the same `aws_ec2` module for the services that can manage AWS instances.
+These two were discarded since they will increase the complexity of the model. Users will need to find all the tables that can contain `instance` information.
+
+Add iasql module's capabilities.
+This option is not viable since it will be really hard to maintain and make it compatible through versions. It will imply the need for several opinionated decisions in behaviour that could not be expected by the users. Also, new users may not understand what capabilities are and add all of them every time.
+
+## Expected Semver Impact
+
+Patch version
+
+## Affected Components
+
+`aws_ec2` module
```

Comment:
I thought it said other modules, such as `aws_vpc` would also be impacted?

Reply:
definitely!